modules/monitoring-setting-query-log-file-for-prometheus.adoc

```diff
@@ -0,0 +1,132 @@
+// Module included in the following assemblies:
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+[id="setting-query-log-file-for-prometheus_{context}"]
+:_content-type: PROCEDURE
+= Enabling the query log file for Prometheus
+
+[role="_abstract"]
+You can configure Prometheus to log all queries run by the engine to a log file.
+
+[IMPORTANT]
+====
+Because log rotation is not supported, only enable this feature temporarily to support troubleshooting.
+====
+
+.Prerequisites
+
+* You have installed the OpenShift CLI (`oc`).
+* *If you are enabling the query log file feature for Prometheus in the `openshift-monitoring` project*:
+** You have access to the cluster as a user with the `cluster-admin` role.
+** You have created the `cluster-monitoring-config` `ConfigMap` object.
+* *If you are enabling the query log file feature for Prometheus in the `openshift-user-workload-monitoring` project*:
+** You have access to the cluster as a user with the `cluster-admin` role, or as a user with the `user-workload-monitoring-config-edit` role in the `openshift-user-workload-monitoring` project.
+** You have created the `user-workload-monitoring-config` `ConfigMap` object.
+
+.Procedure
+
+** *To set the query log file for Prometheus in the `openshift-monitoring` project*:
+. Edit the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
++
+. Add `queryLogFile: <path>` for `prometheusK8s` under `data/config.yaml`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      queryLogFile: <path> <1>
+----
+<1> The full path to the file in which queries will be logged.
++
+. Save the file to apply the changes.
++
+[WARNING]
+====
+When you save changes to a monitoring config map, pods and other resources in the related project might be redeployed. The running monitoring processes in that project might also be restarted.
+====
++
+. Check that the pods for the component are running. The following sample command lists the status of pods in the `openshift-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring get pods
+----
++
+. Read the query log:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring exec prometheus-user-workload-0 -- cat <path>
+----
++
+[IMPORTANT]
+====
+Revert the setting in the config map after you have examined the logged query information.
```

Comment:
The note includes the `===` see the screenshot of the rendered version:
<img width="706" alt="Screen Shot 2022-03-18 at 11 58 39 AM" src="https://user-images.githubusercontent.com/70717303/159038174-b94975b2-2c67-48c9-ad8d-f993a9a7037d.png">